src/main/java/org/osc/controller/nsc/utils/RedirectionApiUtils.java

```diff
@@ -129,9 +131,25 @@ private InspectionPortEntity txInspPortByNetworkElements(NetworkElement ingress,
         String ingressId = ingress != null ? ingress.getElementId() : null;
         String egressId = ingress != null ? egress.getElementId() : null;
 
-        Query q = this.em.createQuery("FROM InspectionPortEntity WHERE ingress_fk = :ingId AND egress_fk = :egId ");
-        q.setParameter("ingId", ingressId);
-        q.setParameter("egId", egressId);
+        CriteriaBuilder cb = this.em.getCriteriaBuilder();
+        CriteriaQuery<InspectionPortEntity> criteria = cb.createQuery(InspectionPortEntity.class);
+        Root<InspectionPortEntity> root = criteria.from(InspectionPortEntity.class);
+
+        SingularAttribute<? super InspectionPortEntity, NetworkElementEntity> ingressAttr =
```

Comment:
This seems way more complicated than it needs to be. Would the below code work? can you update the other places with the same? If this does not work, the original code seems more consise and we can leave it at that.
```
        CriteriaBuilder cb = this.em.getCriteriaBuilder();
        
        CriteriaQuery<InspectionPortEntity> criteria = cb.createQuery(InspectionPortEntity.class);
        Root<InspectionPortEntity> root = criteria.from(InspectionPortEntity.class);
        
        criteria.select(root).where(cb.and(
                cb.equal(root.join("ingressPort").get("elementId"), ingressId),
                cb.equal(root.join("egressPort").get("elementId"), egressId)));
        
        Query q = this.em.createQuery(criteria);
```

Reply:
It actually does result in a double join, but yes, it works.